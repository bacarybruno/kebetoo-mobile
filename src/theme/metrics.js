import { Dimensions } from 'react-native'

const { width, height } = Dimensions.get('window')

const Theme = {
  marginHorizontal: 16,
  marginVertical: 24,
  screenWidth: width,
  screenHeight: height,
  tabBarHeight: 50,
  tabBarFullHeight: 80,
}

export default Theme
