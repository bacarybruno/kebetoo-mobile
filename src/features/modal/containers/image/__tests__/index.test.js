import { Image } from 'react-native'
import { act } from 'react-test-renderer'

import setupTest from 'Kebetoo/src/config/jest-setup'
import metrics from 'Kebetoo/src/theme/metrics'

import ImageModal from '../index'

jest.mock('@react-navigation/native', () => ({
  ...jest.requireActual('@react-navigation/native'),
  useRoute: () => ({
    params: {
      source: {
        uri: 'jest://fake-image.png',
      },
    },
  }),
}))

const givenImageModal = setupTest(ImageModal)({
  navigation: {
    setOptions: jest.fn(),
  },
})

it('renders ImageModal', () => {
  const getImageSizeMock = jest.spyOn(Image, 'getSize')
  getImageSizeMock.mockImplementation((image, onSuccess) => onSuccess(100, 100))

  let wrapper
  act(() => {
    const { wrapper: wrapperAsync } = givenImageModal()
    wrapper = wrapperAsync
  })

  expect(wrapper.toJSON()).toMatchSnapshot()
})

it('renders image without prefedined size', () => {
  const getImageSizeMock = jest.spyOn(Image, 'getSize')
  getImageSizeMock.mockImplementation((image, onSuccess) => onSuccess(1000, 500))

  let wrapper
  act(() => {
    const { wrapper: wrapperAsync } = givenImageModal()
    wrapper = wrapperAsync
  })

  expect(wrapper.root.findAllByType(Image)[0].props.style.aspectRatio).toBe(2)
})

it('renders image with default aspect ratio', () => {
  const getImageSizeMock = jest.spyOn(Image, 'getSize')
  getImageSizeMock.mockImplementation((image, onSuccess, onError) => onError())

  let wrapper
  act(() => {
    const { wrapper: wrapperAsync } = givenImageModal()
    wrapper = wrapperAsync
  })

  expect(wrapper.root.findAllByType(Image)[0].props.style.aspectRatio).toBe(
    metrics.aspectRatio.vertical,
  )
})
