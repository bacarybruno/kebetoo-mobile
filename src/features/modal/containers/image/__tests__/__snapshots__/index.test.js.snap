// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ImageModal 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "rgba(0, 0, 0, 255)",
      "flex": 1,
      "justifyContent": "center",
    }
  }
  testID="pressable"
>
  <View
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "height": 1334,
        "justifyContent": "center",
        "overflow": "hidden",
        "width": 750,
      }
    }
  >
    <View
      renderToHardwareTextureAndroid={true}
      style={
        Object {
          "transform": Array [
            Object {
              "scale": 1,
            },
            Object {
              "translateX": 0,
            },
            Object {
              "translateY": 0,
            },
          ],
        }
      }
    >
      <View
        onLayout={[Function]}
        style={
          Object {
            "height": 750,
            "width": 750,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "jest://fake-image.png",
            }
          }
          style={
            Object {
              "aspectRatio": 1,
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;
